proposed/simplecache.md

```diff
@@ -0,0 +1,207 @@
+    1:35:00.
+
+    Implementing Libraries MAY expire an item before its requested Expiration Time,
+but MUST treat an item as expired once its Expiration Time is reached. If a calling
+library asks for an item to be saved but does not specify an expiration time, or
+specifies a null expiration time or TTL, an Implementing Library MAY use a configured
+default duration. If no default duration has been set, the Implementing Library
+MUST interpret that as a request to cache the item forever, or for as long as the
+underlying implementation supports.
+
+*    **Key** - A string of at least one character that uniquely identifies a
+cached item. Implementing libraries MUST support keys consisting of the
+characters `A-Z`, `a-z`, `0-9`, `_`, and `.` in any order in UTF-8 encoding and a
+length of up to 64 characters. Implementing libraries MAY support additional
+characters and encodings or longer lengths, but must support at least that
+minimum.  Libraries are responsible for their own escaping of key strings
+as appropriate, but MUST be able to return the original unmodified key string.
+The following characters are reserved for future extensions and MUST NOT be
+supported by implementing libraries: `{}()/\@:`
+
+*    **Cache** - An object that implements the `Psr\SimpleCache\CacheInterface` interface.
+
+*    **Cache Misses** - A cache miss will return null and therefore detecting
+if one stored null is not possible. This is the main deviation from PSR-6's
+assumptions.
+
+
+### 1.3 Cache
+
+Implementations MAY provide a mechanism for a user to specify a default TTL
+if one is not specified for a specific cache item.  If no user-specified default
+is provided implementations MUST default to the maximum legal value allowed by
+the underlying implementation.  If the underlying implementation does not
+support TTL, the user-specified TTL MUST be silently ignored.
+
+
+2. Interfaces
+-------------
+
+### 2.1 CacheInterface
+
+The cache interface provides the most basic functionality of cache servers which
+entails basic reading, writing and deleting of single cache items.
+
+In addition it has methods for dealing with multiple sets of cache entries such as writing, reading or
+deleting multiple cache entries at a time. This is useful when you have lots of cache reads/writes
+to perform, and lets you perform your operations in a single call to the cache server cutting down latency
+times dramatically.
+
+Finally for counters it provides the ability to increment and decrement a cache key
+atomically.
+
+```php
+<?php
+
+namespace Psr\SimpleCache;
+
+interface CacheInterface
+{
+    const TTL_MINUTE = 60;
+    const TTL_HOUR = 3600;
+    const TTL_DAY = 86400;
+
+    /**
+     * Fetched a value from the cache.
+     *
+     * @param string $key The unique key of this item in the cache
+     *
+     * @return mixed The value of the item from the cache, or null in case of cache miss
+     */
+    public function get($key);
```

Comment:
From my experience, this should take multiple keys. Whatever implements this interface can handle everything else internally. Otherwise, you will end up with duplicate code that will look pretty bad.

See:

https://github.com/AlekseyKorzun/memcached-wrapper-php/blob/master/src/Memcached/Wrapper.php#L326